src/view/petgraph.rs

```diff
@@ -192,8 +212,65 @@ macro_rules! impl_petgraph_traits {
     };
 }
 
+/// Implement petgraph's `Visitable` and `GetAdjacencyMatrix` traits for a sparse portgraph type or wrapper.
+///
+/// Uses sparse containers to keep track of visited nodes.
+macro_rules! impl_visit_sparse {
+    ($graph:ty) => {impl_visit_sparse!($graph, []);};
+    ($graph:ty, [$($args:tt)*]) => {impl_visit_sparse!($graph, [$($args)*] where );};
+    ($graph:ty, [$($args:tt)*] where $($where:tt)*) => {
+        impl<$($args)*> petgraph::visit::Visitable for $graph where $($where)* {
+            type Map = std::collections::HashSet<Self::NodeId>;
+
+            fn visit_map(&self) -> Self::Map {
+                std::collections::HashSet::new()
+            }
+
+            fn reset_map(&self, map: &mut Self::Map) {
+                map.clear();
+            }
+        }
+
+        impl<$($args)*> petgraph::visit::GetAdjacencyMatrix for $graph where $($where)* {
```

Comment:
I'm surprised there's no `Visitable` dependency for `GetAdjacencyMatrix`, since the first step seems to be reproducing `visit_map` then visiting all of the nodes

Reply:
It's a bit weird because we are doing a "2D visit map" for the matrix.
We use different structures for the 1D and 2D version, instead of something like `Vec<Visitable::Map>`.